File: app/src/main/java/io/github/ovso/righttoknow/ui/map/MapActivity.java
package io.github.ovso.righttoknow.ui.map;

import android.app.FragmentManager;
import android.os.Bundle;
import android.view.MenuItem;
import androidx.annotation.Nullable;
import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.MapFragment;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;
import io.github.ovso.righttoknow.R;
import io.github.ovso.righttoknow.framework.BaseActivity;

/**
 * Created by jaeho on 2018. 3. 15
 */

public class MapActivity extends BaseActivity implements OnMapReadyCallback {
  @Override protected int getLayoutResId() {
    return R.layout.activity_map;
  }

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setTitle(R.string.vio_fac_loc);
    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    FragmentManager fragmentManager = getFragmentManager();
    MapFragment mapFragment = (MapFragment) fragmentManager.findFragmentById(R.id.map_fragment);
    mapFragment.getMapAsync(this);
  }

  @Override public boolean onOptionsItemSelected(MenuItem item) {
    finish();
    return super.onOptionsItemSelected(item);
  }

  @Override public void onMapReady(GoogleMap googleMap) {
    double[] locations = getIntent().getDoubleArrayExtra("locations");
    LatLng latlng = new LatLng(locations[0], locations[1]);
    MarkerOptions markerOptions = new MarkerOptions();
    markerOptions.anchor(0.0f, 1.0f);
    markerOptions.position(latlng);
    markerOptions.title(getIntent().getStringExtra("facName"));
    googleMap.addMarker(markerOptions);
    googleMap.moveCamera(CameraUpdateFactory.newLatLng(latlng));
    googleMap.animateCamera(CameraUpdateFactory.zoomTo(15));
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/ui/map/MapActivity.kt
package io.github.ovso.righttoknow.ui.map

import android.os.Bundle
import android.view.MenuItem
import com.google.android.gms.maps.*
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions
import io.github.ovso.righttoknow.R
import io.github.ovso.righttoknow.databinding.ActivityMapBinding
import io.github.ovso.righttoknow.exts.viewBinding
import io.github.ovso.righttoknow.framework.BaseActivity

class MapActivity : BaseActivity(), OnMapReadyCallback {

  private val binding by viewBinding(ActivityMapBinding::inflate)

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(binding.root)
    setSupportActionBar(toolbar)
    supportActionBar?.setDisplayHomeAsUpEnabled(true)
    setTitle(R.string.vio_fac_loc)
    val mapFragment =
      supportFragmentManager.findFragmentById(R.id.map_fragment) as? SupportMapFragment
    mapFragment?.getMapAsync(this)
    showAd()
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    finish()
    return super.onOptionsItemSelected(item)
  }

  override fun onMapReady(googleMap: GoogleMap) {
    val locations = intent.getDoubleArrayExtra("locations")
    locations?.let {
      val latLng = LatLng(it[0], it[1])
      val markerOptions = MarkerOptions()
      markerOptions.anchor(0.0f, 1.0f)
      markerOptions.position(latLng)
      markerOptions.title(intent.getStringExtra("facName"))
      googleMap.addMarker(markerOptions)
      googleMap.moveCamera(CameraUpdateFactory.newLatLng(latLng))
      googleMap.animateCamera(CameraUpdateFactory.zoomTo(15f))
    }
  }
}
